dvc/lock.py

```diff
@@ -101,44 +148,10 @@ def __del__(self):
                 pass
 
 
-else:
-    import zc.lockfile
-
-    class Lock(object):
-        """Class for dvc repo lock.
-
-        Uses zc.lockfile as backend.
-        """
-
-        def __init__(self, lockfile, tmp_dir=None):
-            self.lockfile = lockfile
-            self._lock = None
-
-        @property
-        def files(self):
-            return [self.lockfile]
-
-        def _do_lock(self):
-            try:
-                self._lock = zc.lockfile.LockFile(self.lockfile)
-            except zc.lockfile.LockError:
-                raise LockError(FAILED_TO_LOCK_MESSAGE)
-
-        def lock(self):
-            try:
-                self._do_lock()
-                return
-            except LockError:
-                time.sleep(DEFAULT_TIMEOUT)
-
-            self._do_lock()
-
-        def unlock(self):
-            self._lock.close()
-            self._lock = None
-
-        def __enter__(self):
-            self.lock()
-
-        def __exit__(self, typ, value, tbck):
-            self.unlock()
+def make_lock(*args, nfs_lock=False, **kwargs):
+    cls = Lock
+    if nfs_lock:
+        if is_py3:
+            raise DvcException("NFS locks are not supported on Python <3.5.")
```

Comment:
@efiop , didn't get this one right away, `if is_py3: not supported on Python <3.5`? woludn't be `if is_py2`?

Reply:
Good catch! Thanks!